filter.go

```diff
@@ -96,7 +96,7 @@ func NewFilterFS(fs FS, opt *FilterOpt) (FS, error) {
 	}
 
 	patternChars := "*[]?^"
-	if os.PathSeparator != '\\' {
+	if filepath.Separator != '\\' {
```

Comment:
Not sure if this can be easily fixed but `FS` should not be platform-specific, so these filters should also not depend on the current os.

Reply:
Hum I think this is based on user input if I'm not mistaken (cc @jedevc)

Reply:
We can leave this for follow-up

Reply:
This seems to come from https://github.com/tonistiigi/fsutil/pull/121.